tree_test.go

```diff
@@ -1982,3 +1982,69 @@ func Benchmark_GetByIndex(b *testing.B) {
 		}
 	})
 }
+
+func TestNodeCacheStatisic(b *testing.T) {
+	db, err := db.NewDB("test", db.MemDBBackend, "")
+	require.NoError(b, err)
+
+	// with cache enable
+	b.Run("with_cache", func(sub *testing.T) {
```

Comment:
Seems like a good candidate to make this table-driven to avoid repetition. Do you mind making it so?

Reply:
Done.

Reply:
Table-driven tests look like the following:
```
func TestSplit(t *testing.T) {
    tests := map[string]struct {
        input string
        sep   string
        expected  []string
    }{
        "simple":       {input: "a/b/c", sep: "/", expected: []string{"a", "b", "c"}},
        "wrong sep":    {input: "a/b/c", sep: ",", expected: []string{"a/b/c"}},
        "no sep":       {input: "abc", sep: "/", expected: []string{"abc"}},
        "trailing sep": {input: "a/b/c/", sep: "/", expected: []string{"a", "b", "c"}},
    }

    for name, tc := range tests {
        t.Run(name, func(t *testing.T) {
            got := Split(tc.input, tc.sep)
            if !reflect.DeepEqual(tc.expected, got) {
                t.Fatalf("expected: %v, got: %v", tc.want, got)
            }
        })
    }
}
```

where there is a `tests` slice and then you iterate over each and run it. Each test case has a name, inputs, and expected values.

Source: https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

A good example in this repo: https://github.com/cosmos/iavl/blob/abaaacdcf919da1fb9fb780862346dcb99b60fd4/internal/encoding/encoding_test.go#L11